query/workqueue.go

```diff
@@ -0,0 +1,56 @@
+package query
+
+// Task is an interface that have a method for returning their index in the
+// work queue.
+type Task interface {
+	// Index returns this Task's index in the work queue.
+	Index() int
```

Comment:
Is the index to convey anything order than a target ordering for consumption?

Reply:
Only priority in which request should be given to peers first